src/app/test.service.ts

```diff
@@ -10,8 +12,7 @@ export class TestService {
   constructor(private http: HttpClient) { }
 
   public getData() {
-    return this.http.get<any>('/assets/files.json')
-      .toPromise()
-      .then(res => <any[]>res.data);
+    return this.http.get('/assets/files.json')
+      .pipe((map((data)=>data)));
```

Comment:
map(data => ObjectMapper.deserializeArray(type, data))

Reply:
Updated the code